Correct Lockdown Mode compile guards to support testing

brentfulgham opened the pull request:
#### 0f236a88c9ed99679cb26a8d19483f9445acc61d
<pre>
Correct Lockdown Mode compile guards to support testing
<a href="https://bugs.webkit.org/show_bug.cgi?id=242867">https://bugs.webkit.org/show_bug.cgi?id=242867</a>

Reviewed by Chris Dumez.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(adjustCoreGraphicsForCaptivePortal):

Canonical link: <a href="https://commits.webkit.org/252586@main">https://commits.webkit.org/252586@main</a>
</pre>

cdumez commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -810,8 +810,6 @@
-#if (!defined(ENABLE_LOCKDOWN_MODE_API) && \
-    (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000 \
-    || PLATFORM(IOS) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 160000))
+#if (!defined(ENABLE_LOCKDOWN_MODE_API) && (PLATFORM(MAC) || PLATFORM(IOS)))
```
Should this be PLATFORM(COCOA) ?

brentfulgham replied:
No -- we don't want this to build on WATCHOS, TVOS, or MACCATALYST.

cdumez reviewed (approved):
OK if it builds on EWS.

brentfulgham committed:
Correct Lockdown Mode compile guards to support testing
https://bugs.webkit.org/show_bug.cgi?id=242867

Reviewed by Chris Dumez.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(adjustCoreGraphicsForCaptivePortal):

Canonical link: https://commits.webkit.org/252586@main
